assignment/week_3/RecommendedMenu/src/components/RecommendMenu.jsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from "react";
+import TasteRecommendation from "./TasteRecommendation";
+import CountDown from "./CountDown";
+
+const RecommendMenu = (props) => {
+  const category = props.category;
+  const step = props.step;
+
+  useEffect(() => {
+    category === "랜덤 추천" ? props.setStep(4) : props.setStep(1);
+  }, []);
+
+  return step === 4 && category === "랜덤 추천" ? (
```

Comment:
애초에 category === "랜덤추천"이어야 step이 4가 되면
```suggestion
  return step === 4  ? (
```
요로케만 해도 대지 않오?

Reply:
일단 요 컴포넌트는 원하는 추천 방식을 누르고 `시작하기` 버튼을 눌렀을 때 동작하는 컴포넌트라서
온보딩 화면에서 `랜덤추천`을 고르고 시작하기를 누른 경우는 step 관련 조건만 있어도 제대로 동작해!

온보딩 화면에서 `취향대로 추천`을 골랐다면 첫 step은 1에서 시작하지만, 마지막에 메뉴를 추천해줄 때는 step 4로 들어가거든..!
이때 여기서 category 관련 조건이 없다면, step이 4이기 때문에 취향대로 추천을 선택했음에도 카운트다운 컴포넌트가 실행되어버려 !
그래서 카테고리 관련 조건까지 추가해서 step이 4라도 취향대로 추천을 진행하고 있다면 Step 컴포넌트로 들어가서 카운트다운 없이 바로 메뉴 추천 결과 화면을 띄워주도록 구현했어 !